src/GraphQL/StringExtensions.cs

```diff
@@ -50,5 +50,51 @@ public static string ToPascalCase(this string s)
 
             return newFirstLetter + s.Substring(1);
         }
+
+        /// <summary>
+        /// Returns a constant case version of this string. For example, converts 'StringError' into 'STRING_ERROR'.
+        /// </summary>
+        public static string ToConstantCase(this string s)
```

Comment:
I hope that the algorithm is correct, but it is complex and I would not like to support it in this form. Also, I don't like the calls to `Substring`.

Reply:
I can eliminate the substring calls easily, but it will make it a little more complex.  If it's the naming you don't like, I can fix that as well.  If you just want to forget the whole thing, that's fine, although this code is 20 times faster than the regex code.  Not just 5% or 10%, but a full 20x improvement, and it does make a measurable difference for scoped schemas as can be seen from the performance results for the introspection and hero tests on scoped schemas.

Reply:
I can also write it using system.buffers if we are going to add that reference.... but it won't make the code any less complex.

Reply:
I can add a whole ton of comments so people know what the code actually does .....

Reply:
I find using `System.Memory` necessary. Comments will be very helpful.

Reply:
Or we leave this algorithm as it is (enriching with comments), given that its results are still cached.